api/features/versioning/serializers.py

```diff
@@ -14,6 +21,34 @@ class Meta(CreateSegmentOverrideFeatureStateSerializer.Meta):
             + ("feature",)
         )
 
+    def save(self, **kwargs):
+        response = super().save(**kwargs)
+
+        feature_state = self.instance
+        if (
+            not feature_state.identity_id
+            and feature_state.feature.external_resources.exists()
+            and feature_state.environment.project.github_project.exists()
+            and feature_state.environment.project.organisation.github_config.exists()
+        ):
+            github_configuration = GithubConfiguration.objects.get(
+                organisation_id=feature_state.environment.project.organisation_id
+            )
+            feature_states = []
+            feature_states.append(feature_state)
+            feature_data: GithubData = generate_data(
+                github_configuration=github_configuration,
+                feature=feature_state.feature,
+                type=WebhookEventType.FLAG_UPDATED.value,
+                feature_states=feature_states,
+            )
+
+            call_github_app_webhook_for_feature_state.delay(
+                args=(asdict(feature_data),),
+            )
+
```

Comment:
Same as [this comment](https://github.com/Flagsmith/flagsmith/pull/3962/files#r1609667423). Factor out this code to a function in `integrations.github` namespace.

Reply:
I created a common function to handle this code.